### PR TITLE
Introduce mappedLocalSubjectMandatory service provider configuration

### PR DESCRIPTION
**Pull request by Yasasr1**
## Purpose
This PR adds the API support for the newly introduced `mappedLocalSubjectMandatory` service provider configuration

### Related carbon-identity-framework PR
- https://github.com/wso2/carbon-identity-framework/pull/5110

### CODE REVIEW

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java [hunk 1]
```diff
@@ -132,6 +132,9 @@ public enum ErrorMessage {
         INVALID_POLICY_TYPE_FOR_API_RESOURCE("60512", "Invalid policy type provided for the API " +
                 "resource.", "API resource with id: %s doesn't allow the provided policy type: %s."),
 
+        ERROR_ASSERT_LOCAL_SUBJECT_IDENTIFIER_DISABLED("60513",
+                "Invalid configuration.",
+                "useMappedLocalSubject cannot be disabled when mappedLocalSubjectMandatory is enabled."),
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java:137]: let's enclose the config names with quotes for better readability

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateClaimConfiguration.java [hunk 2]
```diff
@@ -131,6 +133,14 @@ private void updateSubjectClaimConfigs(SubjectConfig subjectApiModel, ServicePro
 
         if (subjectApiModel != null) {
 
+            if (subjectApiModel.getMappedLocalSubjectMandatory() != null &&
+                    subjectApiModel.getMappedLocalSubjectMandatory() &&
+                    subjectApiModel.getUseMappedLocalSubject() != null &&
+                    !subjectApiModel.getUseMappedLocalSubject()) {
```
Comment (pulasthi7) [inline components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateClaimConfiguration.java:139]: can simplify with Boolean.TRUE.equals() and Boolean.FALSE.equals()

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java (hunk 1 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.common/src/main/java/org/wso2/carbon/identity/api/server/application/management/common/ApplicationManagementConstants.java]: addressed with https://github.com/wso2/identity-api-server/pull/521/commits/c55442cff0175c0b4007f2803256c71b8bcaea36

File: components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateClaimConfiguration.java (hunk 2 above)
Comment (Yasasr1) [reply components/org.wso2.carbon.identity.api.server.application.management/org.wso2.carbon.identity.api.server.application.management.v1/src/main/java/org/wso2/carbon/identity/api/server/application/management/v1/core/functions/application/UpdateClaimConfiguration.java]: addressed with https://github.com/wso2/identity-api-server/pull/521/commits/c55442cff0175c0b4007f2803256c71b8bcaea36

**Review by pulasthi7 — DISMISSED** [review]

**Review by pulasthi7 — APPROVED** [review]

**Pull request merged**